### ISSUE TITLE
fetch() response body does not stream

### ISSUE THREAD

**Issue by sfrieske** [oven-sh/bun#5964]
### What version of Bun is running?

1.0.3+25e69c71e70ac8a0a88f9cf15b4057bd7b2a633a

### What platform is your computer?

Linux 6.1.54-1-MANJARO x86_64 unknown

### What steps can reproduce the bug?

Steps to reproduce:

`var resp = await fetch('http://stream.radioparadise.com/aac-128')`
`var reader = resp.body.getReader()`
`await reader.read()`

### What is the expected behavior?

return `{ value: Uint8Array ..., done: false }`

### What do you see instead?

returns `{ value: undefined, done: true }`

### Additional information

This seems to work on bun with a file URL. 
With an HTTP stream that 'never' ends as above the ReadableStreamDefaultReader (var reader) behaves as though it were closed.

**Comment by jimmywarting** [comment]
Was a bit shocked to see this, Able to reproduce.

**Comment by Electroid** [comment]
It looks like this is happening because the response doesn't have a `Content-Length` or `Transfer-Encoding: chunked` header. This is a bug in Bun, in this case it should read until the connection is closed.

```
< HTTP/1.1 200 OK
< Server: openresty/1.17.8.2
< Date: Sat, 23 Sep 2023 16:10:14 GMT
< Content-Type: audio/aac
< Connection: close
< icy-br: 128
< ice-audio-info: channels=2;samplerate=44100;bitrate=128
< icy-description: DJ-mixed blend of modern and classic rock, electronica, world music, and more. Always 100% commercial-free
< icy-genre: Eclectic
< icy-name: Radio Paradise (128k aac)
< icy-pub: 1
< icy-url: https://radioparadise.com
< Cache-Control: no-cache
< Expires: Sat, 23 Sep 2023 16:10:13 GMT
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Headers: Origin, Accept, X-Requested-With, Content-Type, Icy-MetaData
< Access-Control-Allow-Methods: GET, OPTIONS, SOURCE, PUT, HEAD, STATS
```

**Comment by melroy89** [comment]
It's a streaming data. So expect Bun to keep reading and reading.. Like this:

![stream_data](https://github.com/oven-sh/bun/assets/628926/6bb2bf94-d2a1-4254-b0fc-244e949c5c9e)

Bun will indeed close the connection, just like curl would do. With status 200 OK.

**Comment by jimmywarting** [comment]
> It looks like this is happening because the response doesn't have a Content-Length or Content-Encoding: chunked header. This is a bug in Bun, in this case it should read until the connection is closed.

i think you meant `Transfer-Encoding`

i tried this in chrome and it did stream data.

**Comment by sfrieske** [comment]
Seems to be fixed in v1.0.4, closing.
